plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHttpMetastoreFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.thrift;
+
+import com.google.inject.Inject;
+import io.airlift.units.Duration;
+import io.trino.filesystem.TrinoFileSystemFactory;
+import io.trino.spi.security.ConnectorIdentity;
+
+import java.util.Optional;
+import java.util.concurrent.Executors;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class ThriftHttpMetastoreFactory
+        implements ThriftMetastoreFactory
+{
+    private final IdentityAwareMetastoreClientFactory metastoreClientFactory;
+    private TrinoFileSystemFactory fileSystemFactory;
+
+    @Inject
+    public ThriftHttpMetastoreFactory(
+            TrinoFileSystemFactory fileSystemFactory,
+            IdentityAwareMetastoreClientFactory metastoreClientFactory)
+    {
+        this.metastoreClientFactory = requireNonNull(metastoreClientFactory, "metastoreClientFactory is null");
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+    }
+
+    @Override
+    public boolean isImpersonationEnabled()
+    {
+        return false;
+    }
+
+    @Override
+    public ThriftMetastore createMetastore(Optional<ConnectorIdentity> identity)
+    {
+        return new ThriftHiveMetastore(
+                identity,
+                fileSystemFactory,
+                metastoreClientFactory,
+                2.0,
+                new Duration(1, SECONDS),
+                new Duration(1, SECONDS),
+                new Duration(30, SECONDS),
+                new Duration(10, MINUTES),
+                9,
+                false,
+                false,
+                false,
+                false,
+                true,
+                new ThriftMetastoreStats(),
```

Comment:
please use the same template as in `io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastoreFactory` for the stats

and do expose them for JMX

```
    @Managed
    @Flatten
    public ThriftMetastoreStats getStats()
    {
        return stats;
    }
```